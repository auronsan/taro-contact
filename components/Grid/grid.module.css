.grid-container {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(2, 1fr);
  gap: 10px;

  @media (max-width: 768px) {
    & {
      grid-template-columns: 1fr;
      grid-template-rows: repeat(4, 1fr);
    }
  }
}

.grid-col {
  background-color: var(--color-white);
  padding: 10px;
}
