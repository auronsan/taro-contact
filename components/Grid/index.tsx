import React from 'react';
import clsx from 'clsx';
import { Box } from '@/components/Box';
import classes from './grid.module.css';

export type GridProps = {
  children?: React.ReactNode;
  columns?: number;
  id?: string;
};

const Grid = (props: GridProps) => {
  const { children, columns = 2, id } = props;

  return (
    <Box
      className={clsx(classes['grid-container'], {
        [classes[`grid-container-columns-${columns}`]]: !!columns,
      })}
      id={id}
      data-testid={id}
    >
      {children}
    </Box>
  );
};

export type GridColProps = {
  children?: React.ReactNode;
  id?: string;
};

const GridCol = (props: GridColProps) => {
  const { children, id } = props;
  return (
    <Box className={classes['grid-col']} id={id}>
      {children}
    </Box>
  );
};

Grid.Col = GridCol;

export default Grid;
