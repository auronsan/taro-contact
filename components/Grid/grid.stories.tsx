import Grid from './index';

export default {
  title: 'components/Grid',
  component: Grid,
};

export const Usage = () => (
  <Grid>
    <Grid.Col>Col 1</Grid.Col>
    <Grid.Col>Col 2</Grid.Col>
    <Grid.Col>Col 3</Grid.Col>
  </Grid>
);

export const Usage3Columns = () => (
  <Grid columns={3}>
    <Grid.Col>Col 1</Grid.Col>
    <Grid.Col>Col 2</Grid.Col>
    <Grid.Col>Col 3</Grid.Col>
  </Grid>
);
