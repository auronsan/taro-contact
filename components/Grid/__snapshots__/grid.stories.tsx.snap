// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Grid Usage smoke-test 1`] = `
<main class="roboto-normal">
  <div class="grid_grid-container__wRdc4 grid_grid-container-columns-2__oW8KO">
    <div class="grid_grid-col__PIb_W">
      Col 1
    </div>
    <div class="grid_grid-col__PIb_W">
      Col 2
    </div>
    <div class="grid_grid-col__PIb_W">
      Col 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`components/Grid Usage3Columns smoke-test 1`] = `
<main class="roboto-normal">
  <div class="grid_grid-container__wRdc4 grid_grid-container-columns-3__w_kra">
    <div class="grid_grid-col__PIb_W">
      Col 1
    </div>
    <div class="grid_grid-col__PIb_W">
      Col 2
    </div>
    <div class="grid_grid-col__PIb_W">
      Col 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`grid Usage smoke-test 1`] = `
<main class="roboto-normal">
  <div class="grid_grid-container__wRdc4 grid_grid-container-columns-2__oW8KO">
    <div class="grid_grid-col__PIb_W">
      Col 1
    </div>
    <div class="grid_grid-col__PIb_W">
      Col 2
    </div>
    <div class="grid_grid-col__PIb_W">
      Col 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`grid Usage3Columns smoke-test 1`] = `
<main class="roboto-normal">
  <div class="grid_grid-container__wRdc4 grid_grid-container-columns-3__w_kra">
    <div class="grid_grid-col__PIb_W">
      Col 1
    </div>
    <div class="grid_grid-col__PIb_W">
      Col 2
    </div>
    <div class="grid_grid-col__PIb_W">
      Col 3
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       style="padding: 0px;"
  >
  </div>
</main>
`;
